.github/workflows/release.yml

```diff
@@ -5,22 +5,136 @@ on:
     tags:
       - '*'
 
+
 jobs:
-  goreleaser:
-    runs-on: ubuntu-latest
+  prepare-windows:
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
     steps:
       - uses: actions/checkout@v3
+      - uses: msys2/setup-msys2@v2
         with:
-          fetch-depth: 0
+          msystem: UCRT64
+          update: true
+          install: git mingw-w64-ucrt-x86_64-gcc
+      - shell: bash
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+      - id: cache
+        uses: actions/cache@v3
+        with:
+          path: dist/windows
+          key: windows-${{ env.sha_short }}
       - uses: actions/setup-go@v4
         with:
           go-version-file: 'go.mod'
-      - name: Install GoReleaser
+      - name: Release
+        if: steps.cache.outputs.cache-hit != 'true'
         uses: goreleaser/goreleaser-action@v6
         with:
-          install-only: true
           version: latest
+          distribution: goreleaser-pro
+          args: release --clean --split
+        env:
+          GITHUB_TOKEN: ${{ secrets.RELEASE_TOKEN }}
+          GORELEASER_KEY: ${{ secrets.GORELEASER_KEY }}
+          VERSION: ${{ github.ref_name }}
+          GGOOS: windows
+
+  prepare-darwin:
```

Comment:
for after we merge this: could we make the `prepare-darwin` and `prepare-linux` ones to be a matrix-build instead, since the blocks seem almost the same?